core/src/test/scala/unit/kafka/network/SocketServerTest.scala

```diff
@@ -316,6 +313,25 @@ class SocketServerTest extends JUnitSuite {
     }
   }
 
+  private def sendRequestsUntilStagedReceive(server: SocketServer, socket: Socket, requestBytes: Array[Byte]): RequestChannel.Request = {
+    def sendTwoRequestsReceiveOne(): RequestChannel.Request = {
+      sendRequest(socket, requestBytes, flush=false)
+      sendRequest(socket, requestBytes, flush=true)
+      server.requestChannel.receiveRequest(2000)
+    }
+    val (request, hasStagedReceives) = TestUtils.computeUntilTrue(sendTwoRequestsReceiveOne()) { req =>
+      val connectionId = req.connectionId
+      val hasStagedReceives = server.processor(0).numStagedReceives(connectionId) > 0
+      if (!hasStagedReceives) {
+        processRequest(server.requestChannel, req)
+        processRequest(server.requestChannel)
+      }
+      hasStagedReceives
+    }
+    assertTrue("Receives not staged", hasStagedReceives)
```

Comment:
nit: isn't this a bit of a tautology? The previous statement will only complete if `hasStagedReceives` is true.

Reply:
Maybe mention the timeout in the error message?